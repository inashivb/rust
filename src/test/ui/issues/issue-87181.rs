trait Foo {
    fn get(&self);
}

struct Bar<T>(T);

struct Inner();

impl Foo for Inner {
    fn get(&self) { }
}

fn main() {
    let thing = Bar(Inner);
    thing.0.get();
    //~^ ERROR no method named `get` found for fn item `fn() -> Inner {Inner}` in the current scope
}
