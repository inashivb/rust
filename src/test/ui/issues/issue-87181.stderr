error[E0599]: no method named `get` found for fn item `fn() -> Inner {Inner}` in the current scope
  --> $DIR/issue-87181.rs:15:13
   |
LL |     thing.0.get();
   |             ^^^ method not found in `fn() -> Inner {Inner}`
   |
   = note: `thing.0` is a function, perhaps you wish to call it
   = help: items from traits can only be used if the trait is implemented and in scope
note: `Foo` defines an item `get`, perhaps you need to implement it
  --> $DIR/issue-87181.rs:1:1
   |
LL | trait Foo {
   | ^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
